pkg/logentry/stages/timestamp.go

```diff
@@ -158,6 +175,16 @@ func (ts *timestampStage) Process(labels model.LabelSet, extracted map[string]in
 		return
 	}
 
+	if ts.lastKnownTimestamps != nil {
+		labelsStr := labels.String()
+		lastTimestamp, ok := ts.lastKnownTimestamps.Get(labelsStr)
+		if ok && (lastTimestamp.(time.Time).Equal(*parsedTs) ||
```

Comment:
I don't think equality causes an issue ? no ?

Reply:
Tbh I don't know anymore. I had a ton of "out of order" errors which this seemed to solve.

But now I've given up on using loki altogether, when I tried ingesting old files using --stdin it processed like 5 files then froze without any output or info, so I just gave up